autofit/non_linear/search/nest/nautilus/search.py

```diff
@@ -0,0 +1,342 @@
+
+        Returns
+        -------
+        A result object comprising the Samples object that includes the maximum log likelihood instance and full
+        set of accepted ssamples of the fit.
+        """
+
+        try:
+            from nautilus import Sampler
+        except ModuleNotFoundError:
+            raise ModuleNotFoundError(
+                "\n--------------------\n"
+                "You are attempting to perform a model-fit using Nautilus. \n\n"
+                "However, the optional library Nautilus (https://nautilus-sampler.readthedocs.io/en/stable/index.html) is "
+                "not installed.\n\n"
+                "Install it via the command `pip install nautilus-sampler==0.7.2`.\n\n"
+                "----------------------"
+            )
+
+        fitness = self.Fitness(
+            model=model,
+            analysis=analysis,
+            log_likelihood_cap=log_likelihood_cap,
+        )
+
+        if conf.instance["non_linear"]["nest"][self.__class__.__name__][
+            "parallel"
+        ].get("force_x1_cpu") or self.kwargs.get("force_x1_cpu"):
+            pool = None
+        else:
+            pool = self.number_of_cores
+
+        if conf.instance["non_linear"]["nest"][self.__class__.__name__][
+            "parallel"
+        ].get("force_x1_cpu") or self.kwargs.get("force_x1_cpu"):
+
+            sampler = Sampler(
+                prior=prior_transform,
+                likelihood=fitness.__call__,
+                n_dim=model.prior_count,
+                prior_kwargs={"model": model},
+                filepath=self.paths.search_internal_path / "checkpoint.hdf5",
+                pool=None,
+                **self.config_dict_search
+            )
+
+            sampler.run(
+                **self.config_dict_run,
+            )
+
+        elif not self.mpi:
+
+    
+            sampler = Sampler(
+                prior=prior_transform,
+                likelihood=fitness.__call__,
+                n_dim=model.prior_count,
+                prior_kwargs={"model": model},
+                filepath=self.paths.search_internal_path / "checkpoint.hdf5",
+                pool=self.number_of_cores,
+                **self.config_dict_search
+            )
+
+            sampler.run(
+                **self.config_dict_run,
+            )
+
+        elif self.mpi:
+
+            from mpi4py import MPI
+            comm = MPI.COMM_WORLD
+
+            logger.info(f"Search beginning with MPI {comm.Get_rank()} / {self.number_of_cores}")
+
+            from mpi4py.futures import MPIPoolExecutor
+            pool = MPIPoolExecutor(self.number_of_cores)
+
+            sampler = Sampler(
+                prior=prior_transform,
+                likelihood=fitness.__call__,
+                n_dim=model.prior_count,
+                prior_kwargs={"model": model},
+                filepath=self.paths.search_internal_path / "checkpoint.hdf5",
+                pool=pool,

```

Comment:
Not sure if I'm nmissing somethign but it looks like only pool changes in value? In which case use the if-else to create the pool object and don't repeat all the other lines instantiating the sampler.